client/resource_group/controller/limiter.go

```diff
@@ -406,7 +406,9 @@ func (lim *Limiter) reserveN(now time.Time, n float64, maxFutureReserve time.Dur
 	}
 	// Update state
 	if ok {
-		lim.last = now
+		if lim.last.Before(now) {
+			lim.last = now
+		}
```

Comment:
We better extract a function to update `lim.last` to force the check each time we try to update it. This should prevent this problem from being reintroduced again in the future.